main.cpp

```diff
@@ -51,42 +51,216 @@ struct CarWash            //1) a U.D.T. with a random number of member variables
 /*
  1)
  */
+struct Person
+{
+    int numHands = 2;
+    float height = 1.68f; // in [m]
+    
+    struct Hand
+    {
+        bool isLeftHand = false;
+        float fingerLength = 0.082f;        //2) in [m]
+    };
+    
+    void isMale( Person person );
+    
+    Hand myLeftHand;
+};
 
 /*
  2)
  */
+struct Computer
+{
+    int numKeys = 48;
+    float processorSpeed = 4.2f; // in [GHz]
+    bool isLaptop = true;
+
+    struct App
+    {
+        bool isStandalone = true;
+        float appSize = 300.0f; // in [MB]
+    };
+    
+    void formatComputer(bool toFormat);
+    
+    App JUCE;
+};
+
 
 /*
  3)
  */
+struct Watch
+{
+    int numBatteries = 1;
+    int totalHours = 12;
+    float weight = 50.0f; // in [g]
+    bool isDigital = true;
+
+    struct Wristband
+    {
+        bool isLeather = true;
+        int color = 25; // we don't know vectors so I'm supposing a unidimensional color scale :P
+    };
+    
+    void reset(bool toReset);
+    void setAlarm(float timeAlarm);
+    
+    Wristband myPlasticWB;
+}; 
 
 /*
  4)
  */
+struct Window
+{
+    float height = 2.0f; // in [m]
+    float width = 1.0f; // in [m]
+    float weight = 20.0f; // in [kg]
+    bool isClean = true;
+    bool isGlassOpaque = true;
+
+    struct Knob
+    {
+        bool isSteel = true;
+        int color = 5; // we don't know vectors so I'm supposing a unidimensional color scale :P
+    };
+    
+    void cleanWindow(bool toClean);
+    void openWindon(bool toOpen);
+    
+    Knob myKnob;
+}; 
 
 /*
  5)
  */
+struct Table
+{
+    int numLegs = 4;
+    float height = 2.0f; // in [m]
+    float width = 1.0f; // in [m]
+    float weight = 20.0f; // in [kg]
+    bool isWooden = true;
+
+    struct Tablecloth
+    {
+        bool isPlastic = true;
+        float height = 1.5f; // in [m]
+        float width = 1.0f; // in [m]
+    };
+    
+    void cleanTable(bool toClean);
+    void placeTablecloth(Tablecloth myTablecloth);
+    
+    Tablecloth myTablecloth;
+}; 
 
 /*
  6)
  */
+struct Train
+{
+    int numCarriages = 20;
+    float height = 10.0f; // in [m]
+    float length = 10.0f; // in [m]
+    float width = 3.0f; // in [m]
+    float weightPerCarriage = 10.0f; // in [ton]
+    float maxSpeed = 75; // in [km/h]
+
+    struct Carriage
+    {
+        int numWindows = 10;
+        bool isSeatLeather = true;
+        int maxNumPassengers = 25;
+    };
+    
+    void start();
+    void stop();
+    
+    Carriage myCarriage;
+};  
 
 /*
  7)
  */
+struct Shoe
+{
+    struct Lace
+    {
+        float length = 2; // in [m]
+        int color = 25;
+    };
+    
+    struct Logo
+    {
+        int color = 25;
+        int logoIndex = 2; // assuming a preexisting vector including all the logos
+    };
+
```

Comment:
Can you add a member function or two to this? 

- [ ] add member func declarations